examples/mm/stable_diffusion/train_sd_draftp.py

```diff
@@ -0,0 +1,149 @@
+def main(cfg) -> None:
+
+    logging.info("\n\n************** Experiment configuration ***********")
+    logging.info(f"\n{OmegaConf.to_yaml(cfg)}")
+
+    # TODO: has to be set true for PyTorch 1.12 and later.
+    torch.backends.cuda.matmul.allow_tf32 = True
+    cfg.model.data.train.dataset_path = [cfg.model.data.webdataset.local_root_path for _ in range(cfg.trainer.devices)]
+    trainer = MegatronStableDiffusionTrainerBuilder(cfg).create_trainer()
+    exp_manager(trainer, cfg.exp_manager)
+    logger = CustomLoggerWrapper(trainer.loggers)
+    ptl_model = load_from_nemo(MegatronSDDRaFTPModel, cfg.model, trainer).to(torch.cuda.current_device())
+    # TODO: @geshen: Check why we have PEFT init here
+    if cfg.model.get("peft", None):
+        if cfg.model.peft.enable:
+            peft_cfg_cls = PEFT_CONFIG_MAP[cfg.model.peft.peft_scheme]
+
+            if cfg.model.peft.restore_from_path is not None:
+                # initialize peft weights from a checkpoint instead of randomly
+                # This is not the same as resume training because optimizer states are not restored.
+                logging.info("PEFT Weights will be loaded from", cfg.model.peft.restore_from_path)
+                ptl_model.load_adapters(cfg.model.peft.restore_from_path, peft_cfg_cls(cfg.model))
+
+            elif peft_cfg_cls is not None:
+                logging.info("Adding adapter weights to the model for PEFT")
+                ptl_model.add_adapter(peft_cfg_cls(cfg.model))
+            else:
+                logging.info(f"Running full finetuning since no peft scheme is given.\n{ptl_model.summarize()}")
+
+    with open_dict(cfg):
+        cfg.model.precision = cfg.trainer.precision
+
+    trainer_restore_path = trainer.ckpt_path
+
+    if trainer_restore_path is not None:
+        custom_trainer_state_dict = retrieve_custom_trainer_state_dict(trainer)
+        consumed_samples = custom_trainer_state_dict["consumed_samples"]
+    else:
+        custom_trainer_state_dict = None
+        consumed_samples = 0
+
+    init_distributed(trainer, ptl_model, cfg.model.get("transformer_engine", False))
+
+    train_dataset, _ = text_webdataset.build_train_valid_datasets(cfg.model, consumed_samples=consumed_samples)
+    train_dataset = [d["captions"] for d in list(train_dataset)]
+
+    train_dataloader = build_dataloader(
+        cfg,
+        dataset=train_dataset,
+        consumed_samples=consumed_samples,
+        mbs=cfg.model.micro_batch_size,
+        gbs=cfg.model.global_batch_size,
+        load_gbs=True,
+    )
+
+    if cfg.model.get("transformer_engine", False):
+        ptl_model.setup_transformer_engine_tp_groups()
+
+    ptl_model.setup()
+    trainer.strategy._lightning_module = ptl_model
+
+    dummy_train_dataloader = torch.utils.data.DataLoader(
+        dataset=train_dataset,
+        batch_size=divide(cfg.model.global_batch_size, parallel_state.get_data_parallel_world_size()),
+    )
```

Comment:
this is only needed if your train step is wrong using the train_dataloader, which is the case in PPO but is it the case here? for example in SFT we don't need this ugly hack